design/metastore-project.md

```diff
@@ -0,0 +1,19 @@
+# Next Generation Metastore - Project
+
+
+## Milestone 1 - lakeFS Metastore Proposal
+
+Items for discover listed with hierarchy as dependency:
+
+- Passing reference/branch information
+    - Authentication with lakeFS
+    - Side by side - very similar to the current work on side-by-side work with lakeFS and the current object-storage, the user like to run the same application with minimum changes, while enabling some of the data to live inside lakeFS. What needs to change or requirements in order to have the same in the metadata level? Do we need to sync information, manage part of the data and enable mapping on the client side? Can we replace the client side implementation? do we need to spin different endpoints to serve different repositories, branches in lakeFS for parts of the metadata? fallback to existing metastore?
+- Data model - Graveler to enable 3-way diff / merge that will work in optimal way.
```

Comment:
Please clarify how this item is different from the 3-way diff that Graveler supports today.

Reply:
Graveler works on a entry level - each entry change will appear as added/removed/changed.
The data model holds different entries with relations - example: partition is an entity you can append to a table. if we model the partition is an entry in graveler a diff between the two will not explain to the end-user what what database/table was updated. Also without understanding the change it may render as conflict if both sides of the merge will append a partition, while we like to merge both of them.